videosink/display.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// The videosink package provides a display driver implementing an HTTP request
+// handler. Client requests get an initial snapshot of the graphics buffer and
+// are updated further on every change.
+//
+// The primary use case is the development of display outputs on a host
+// machine. Additionally devices with network connectivity can use this driver
+// to provide a copy of their local display via a web interface.
+//
+// The protocol used is "MJPEG" (https://en.wikipedia.org/wiki/Motion_JPEG)
+// which is often used by IP cameras. Because of its better suitability for
+// computer-drawn graphics the PNG image format is used by default. JPEG as
+// a format can be selected via Options.Format or using the "format" URL
+// parameter.
+package videosink
+
+import (
+	"image"
+	"image/color"
+	"image/draw"
+	"net/http"
+	"sync"
+
+	"periph.io/x/conn/v3/display"
+)
+
+// Options for videosink devices.
+type Options struct {
+	// Width and height of the image buffer.
+	Width, Height int
+
+	// Format specifies the image format to send to clients.
+	Format ImageFormat
+}
+
+type Display struct {
```

Comment:
I don't think Display is a good name, also document.

Reply:
Do you have a better name in mind? `Drawer` as in my first round, or maybe `Sink`? `videosink.VideoSink` reads like [unnecessary stuttering](https://stackoverflow.com/a/45886055).

Reply:
Technically, it's a stream of images, not really a video. images.Sink is a possibility.

My main concern is that it's really not a display. :) I agree we should try to not stutter.

Reply:
You're right that it's currently a stream of images. Technically it would be possible to plug in a video encoder and stream the result of that to the HTTP client. I didn't do this because there's no video codec encoder in the standard library and images should be fine for development.

To be honest I'm not a fan of `Sink`. The _thing_ is also a virtual device, so the structure could be called `Device`. On the other hand it's an actual HTTP request handler masquerading as a display driver for periph.io, so `Handler` would also fit.

Which would you prefer?

Naming is hard.

Reply:
I think I prefer Device to Display but it's a bit misleading. Given it's misleading, I'm fine with sticking to Display and moving on.